cpg_infra/driver.py

```diff
@@ -1075,10 +1075,7 @@ def setup_storage_archive_bucket_permissions(self):
     def archive_bucket(self):
         return self.infra.create_bucket(
             'archive',
-            lifecycle_rules=[
-                self.infra.bucket_rule_archive(days=self.dataset_config.archive_age),
-                self.infra.bucket_rule_undelete(),
-            ],
+            lifecycle_rules=[self.infra.bucket_rule_undelete()],
```

Comment:
How does this actually be in the cold tier if autoclass is false and tehre are no lifecycle rules or bucket tier options?

Reply:
Of course you're completely right, this PR was pretty nonsensical. It's easy to change the `storage_class` of a bucket in GCS (https://www.pulumi.com/registry/packages/gcp/api-docs/storage/bucket/#inputs), but unfortunately that concept doesn't seem to exist in Azure, as far as I can tell (https://learn.microsoft.com/en-us/azure/storage/blobs/access-tiers-overview#changing-a-blobs-access-tier).

Hence I'll close this PR and instead submit one where we default to an archive age / grace period of 0 days.